src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs

```diff
@@ -106,215 +111,210 @@ private IScriptExecutor ScriptExecutor
+                            && !possibleSdkPath.EndsWith(".dll", StringComparison.OrdinalIgnoreCase)
+                            && Directory.Exists(possibleSdkPath))
                         {
-                            var pathToProject = GetPathToDTEProjectLookup(project);
+                            var possibleProjectJson = Path.Combine(
+                                                        possibleSdkPath,
+                                                        BuildIntegratedProjectUtility.ProjectConfigFileName);
 
-                            foreach (var xProjPath in xprojFiles)
+                            if (File.Exists(possibleProjectJson))
                             {
-                                // Only add projects that we can find in the solution, otherwise they will
-                                // end up causing failures. If this is an actual failure the resolver will
-                                // fail when resolving the dependency from project.json
-                                Project xProjDTE;
-                                if (pathToProject.TryGetValue(xProjPath, out xProjDTE))
-                                {
-                                    var xProjFullPath = EnvDTEProjectUtility.GetFullProjectPath(xProjDTE);
-                                    childReferences.Add(xProjFullPath);
-
-                                    // Continue walking this project if it has not been walked already
-                                    if (!uniqueProjects.Contains(xProjFullPath))
-                                    {
-                                        toProcess.Enqueue(xProjDTE);
-                                    }
-                                }
+                                childReferences.Add(possibleProjectJson);
+
+                                // add the sdk to the results here
+                                result.Processed.Add(new ExternalProjectReference(
+                                    possibleProjectJson,
+                                    childReference.Name,
+                                    possibleProjectJson,
+                                    msbuildProjectPath: null,
+                                    projectReferences: Enumerable.Empty<string>()));
                             }
                         }
                     }
+                }
+                catch (Exception ex)
+                {
+                    // Exceptions are expected in some scenarios for native projects,
+                    // ignore them and show a warning
+                    hasMissingReferences = true;
 
-                    results.Add(new ExternalProjectReference(
-                        projectFileFullPath,
-                        packageSpec,
-                        projectFileFullPath,
-                        childReferences));
+                    Debug.Fail("Unable to find project closure: " + ex.ToString());
```

Comment:
It's logged as a warning. It's potentially serious since it could mean that a project reference is missing.

https://github.com/NuGet/NuGet.Client/blob/emgarten/dtep2p/src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs#L275